### PR TITLE
Remove deprecated 'PROPERTY_BASED_DESCRIPTORS'

### PR DESCRIPTION
**Pull request by btecu**

### CODE REVIEW

File: packages/@ember/-internals/metal/lib/property_get.ts [hunk 1]
```diff
@@ -117,29 +116,7 @@ export function get(obj: object, keyName: string): any {
...
-      );
-
-      meta(obj).writeDescriptors(keyName, value);
-
-      value.setup(obj, keyName);
-
-      return value.get(obj, keyName);
-    }
-  } else {
     value = obj[keyName];
```
Comment (rwjblue) [inline packages/@ember/-internals/metal/lib/property_get.ts]: Should be dedented right?

Comment (btecu) [reply packages/@ember/-internals/metal/lib/property_get.ts]: Yes, fixed it.

**Comment by btecu** [comment]
It probably closes #15589.

**Review by auvipy — APPROVED** [review]

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you @btecu!

**Pull request merged**